cloud/pkg/cloudhub/handler/messagehandler.go

```diff
@@ -445,30 +466,12 @@ func (mh *MessageHandle) MessageWriteLoop(info *model.HubInfo, stopServe chan Ex
 		copyMsg := deepcopy(msg)
 		trimMessage(copyMsg)
 
-		// initialize timer and retry count for sending message
-		var (
-			retry                       = 0
-			retryInterval time.Duration = 2
-		)
-
-		for {
-			conn, ok := mh.nodeConns.Load(info.NodeID)
-			if !ok {
-				if retry == 1 {
-					break
-				}
-				retry++
-				time.Sleep(time.Second * retryInterval)
-				continue
-			}
-			err := mh.sendMsg(conn.(hubio.CloudHubIO), info, copyMsg, msg, nodeStore)
-			if err != nil {
-				klog.Errorf("Failed to send event to node: %s, affected event: %s, err: %s",
-					info.NodeID, dumpMessageMetadata(copyMsg), err.Error())
-				nodeQueue.AddRateLimited(key.(string))
-				time.Sleep(time.Second * 2)
-			}
-			break
+		err := mh.sendMsg(conn.(hubio.CloudHubIO), info, copyMsg, msg, nodeStore)
+		if err != nil {
+			klog.Errorf("Failed to send event to node: %s, affected event: %s, err: %s",
+				info.NodeID, dumpMessageMetadata(copyMsg), err.Error())
+			nodeQueue.Add(key.(string))
```

Comment:
I think can we add another judgement here, like:
```
if strings.Contains(err.Error(), "use of closed network connection") {
		mh.nodeConns.Delete(info.NodeID)
                continue
}
```
To avoid print too many err logs when edgenode offline, and the keepalive still haven't detect it. 
And the `use of closed network connection` seems should be a standard out put. Ref: https://github.com/kubeedge/kubeedge/blob/master/vendor/golang.org/x/net/http2/server.go#L674
@My-pleasure

Reply:
the err logs:
```
E0210 21:14:43.231232 2402450 messagehandler.go:471] Failed to send event to node: edge-node, affected event: id: 2c9c451f-7c21-40f6-bceb-552d8e288d56, parent_id: , group: resource, source: edgecontroller, resource: default/pod/nginx-deployment-7d5bf4bc7-rlgqn, operation: update, err: use of closed network connection
E0210 21:14:45.231488 2402450 messagehandler.go:471] Failed to send event to node: edge-node, affected event: id: 2c9c451f-7c21-40f6-bceb-552d8e288d56, parent_id: , group: resource, source: edgecontroller, resource: default/pod/nginx-deployment-7d5bf4bc7-rlgqn, operation: update, err: use of closed network connection
E0210 21:14:47.231661 2402450 messagehandler.go:471] Failed to send event to node: edge-node, affected event: id: 2c9c451f-7c21-40f6-bceb-552d8e288d56, parent_id: , group: resource, source: edgecontroller, resource: default/pod/nginx-deployment-7d5bf4bc7-rlgqn, operation: update, err: use of closed network connection
E0210 21:14:49.231859 2402450 messagehandler.go:471] Failed to send event to node: edge-node, affected event: id: 2c9c451f-7c21-40f6-bceb-552d8e288d56, parent_id: , group: resource, source: edgecontroller, resource: default/pod/nginx-deployment-7d5bf4bc7-rlgqn, operation: update, err: use of closed network connection
E0210 21:14:51.232059 2402450 messagehandler.go:471] Failed to send event to node: edge-node, affected event: id: 2c9c451f-7c21-40f6-bceb-552d8e288d56, parent_id: , group: resource, source: edgecontroller, resource: default/pod/nginx-deployment-7d5bf4bc7-rlgqn, operation: update, err: use of closed network connection
E0210 21:14:53.232728 2402450 messagehandler.go:471] Failed to send event to node: edge-node, affected event: id: 2c9c451f-7c21-40f6-bceb-552d8e288d56, parent_id: , group: resource, source: edgecontroller, resource: default/pod/nginx-deployment-7d5bf4bc7-rlgqn, operation: update, err: use of closed network connection
E0210 21:14:55.233382 2402450 messagehandler.go:471] Failed to send event to node: edge-node, affected event: id: 2c9c451f-7c21-40f6-bceb-552d8e288d56, parent_id: , group: resource, source: edgecontroller, resource: default/pod/nginx-deployment-7d5bf4bc7-rlgqn, operation: update, err: use of closed network connection
E0210 21:14:57.234131 2402450 messagehandler.go:471] Failed to send event to node: edge-node, affected event: id: 2c9c451f-7c21-40f6-bceb-552d8e288d56, parent_id: , group: resource, source: edgecontroller, resource: default/pod/nginx-deployment-7d5bf4bc7-rlgqn, operation: update, err: use of closed network connection

```

Reply:
Good idea